ecc/bn254/fr/poseidon/constants/constants.go

```diff
@@ -0,0 +1,61 @@
+package constants
+
+import (
+	"math/big"
+
+	"github.com/consensys/gnark-crypto/ecc/bn254/fr"
+)
+
+// Number of full rounds
+const RF = 8
+
+// Generated by https://extgit.iaik.tugraz.at/krypto/hadeshash/-/blob/master/code/calc_round_numbers.py
+// And rounded up to nearest integer that divides by t in [3, 13]
+var RP = []int{57, 56, 60, 60, 63, 64, 63, 60, 66, 60, 65}
+
+// Round constants
+var RC [][]*fr.Element
+
+// Maximum distance separable matrix
+var MDS [][][]*fr.Element
+
+var Alpha *big.Int
+
+func init() {
+	// Parameters are generated by a reference script https://extgit.iaik.tugraz.at/krypto/hadeshash/-/blob/master/code/generate_parameters_grain.sage
+	// Used like so: sage generate_parameters_grain.sage 1 0 254 2 8 57 0x30644e72e131a029b85045b68181585d2833e84879b9709143e1f593f0000001
+	constantsSTR := [][]string{constantsT3, constantsT4, constantsT5, constantsT6, constantsT7, constantsT8, constantsT9, constantsT10, constantsT11, constantsT12, constantsT13}
+	mdsSTR := [][][]string{mdsT3, mdsT4, mdsT5, mdsT6, mdsT7, mdsT8, mdsT9, mdsT10, mdsT11, mdsT12, mdsT13}
+	size := len(RP)
+	RC = make([][]*fr.Element, size)
+	MDS = make([][][]*fr.Element, size)
+	Alpha = big.NewInt(5)
```

Comment:
some of the sbox using params as 7, would it less security by reduce to 5 ?

Reply:
Alpha is the smallest integer such that gcd(alpha, p) != 1. So alpha= 3, 5, and 7 are all ok but choosing 7 means more constraints than 5. Also, in most implementations, you would find the value alpha=5 
https://www.poseidon-hash.info/